### PR TITLE
Коньков Иван. Задача 3. Вариант 30. "Повышение контраста полутонового изображения посредством линейной растяжки гистограммы"

### PR DESCRIPTION
**Pull request by keshaproger**
Алгоритм:

Нахождение min и max: Определить минимальное и максимальное значения яркости во входном изображении.

Линейное преобразование: Для каждого пикселя применить формулу:
output = ((input - min) / (max - min)) * 255
Если все пиксели одинаковы (min = max), оставить изображение без изменений.

Особенности:

Обработка граничных случаев (одинаковые значения, уже полный диапазон).

Валидация входных данных (размеры массивов, указатели).

Поддержка больших данных (тесты на производительность с 100 млн элементов).

Тесты:

Базовый: Проверка полного диапазона входных данных (0→0, 255→255).

Константа: Все пиксели одинаковы — выход равен входу.

Кастомный диапазон: Проверка корректности растяжения (50→0, 200→255).

Производительность: Замер времени для оптимизации (PipelineRun и TaskRun).

### CODE REVIEW

File: tasks/mpi/konkov_i_linear_hist_stretch_mpi/data/test.txt [hunk 1]
```diff
@@ -0,0 +1 @@
+100
```
Comment (aobolensk) [inline tasks/mpi/konkov_i_linear_hist_stretch_mpi/data/test.txt:1]: Do you need this?

File: tasks/mpi/konkov_i_linear_hist_stretch_mpi/func_tests/main.cpp [hunk 2]
```diff
@@ -0,0 +1,68 @@
+#include <gtest/gtest.h>
+
+#include <cstdint>
+#include <memory>
+#include <vector>
+
+#include "core/task/include/task.hpp"
+#include "mpi/konkov_i_linear_hist_stretch_mpi/include/ops_mpi.hpp"
+
```
Comment (aobolensk) [inline tasks/mpi/konkov_i_linear_hist_stretch_mpi/func_tests/main.cpp:9]: Please, add test cases for corner cases like empty image

**Review by aobolensk — CHANGES_REQUESTED** [review]

File: tasks/mpi/konkov_i_linear_hist_stretch_mpi/data/test.txt (hunk 1 above)
Comment (keshaproger) [reply tasks/mpi/konkov_i_linear_hist_stretch_mpi/data/test.txt]: deleted

File: tasks/mpi/konkov_i_linear_hist_stretch_mpi/func_tests/main.cpp (hunk 2 above)
Comment (keshaproger) [reply tasks/mpi/konkov_i_linear_hist_stretch_mpi/func_tests/main.cpp:9]: added

**Review by aobolensk — APPROVED** [review]
